src/solutions/pde_solutions.jl

```diff
@@ -0,0 +1,190 @@
+"""
+$(TYPEDEF)
+
+Solution to a PDE, solved from an ODEProblem generated by a discretizer.
+
+## Fields
+- `u`: the solution to the PDE, as a dictionary of symbols to Arrays of values. The
+  Arrays are of the same shape as the domain of the PDE. Time is always the first axis.
+- `original_sol`: The original ODESolution that was used to generate this solution.
+- `t`: the time points corresponding to the saved values of the ODE solution.
+- `ivgrid`: The full list of axies for the solution
+- `ivs`: The list of independent variables for the solution.
+- `dvs`: The list of dependent variables for the solution.
+- `disc_data`: Metadata about the discretization process and type.
+- `prob`: The ODEProblem that was used to generate this solution.
+- `alg`: The algorithm used to solve the ODEProblem.
+- `interp`: Interpolations for the solution.
+- `retcode`: The return code from the solver. Used to determine whether the solver solved
+  successfully (`sol.retcode === :Success`), whether it terminated due to a user-defined
+  callback (`sol.retcode === :Terminated`), or whether it exited due to an error. For more
+  details, see the return code section of the ODEProblem.jl documentation.
+"""
+struct PDESolution{T, N, uType, Disc, Sol, DType, tType, gridType, ivType, dvType, P, A,
+                   IType} <: AbstractPDESolution{T, N, uType}
+    u::uType
+    original_sol::Sol
+    errors::DType
+    t::tType
+    ivgrid::gridType
+    ivs::ivType
+    dvs::dvType
+    disc_data::Disc
+    prob::P
+    alg::A
+    interp::IType
+    dense::Bool
+    tslocation::Int
+    retcode::Symbol
+end
+
+function PDESolution(sol::ODESolution{T}, metadata::MOLMetadata) where {T}
+    odesys = sol.prob.f.sys
+
+    pdesys = metadata.pdesys
+    discretespace = metadata.discretespace
+
+    ivs = [discretespace.time, discretespace.x̄...]
+    ivgrid = (isequal(discretespace.time, x) ? sol.t : discretespace.grid[x] for x in ivs)
+    # Reshape the solution to flat arrays
+    umap = Dict(map(discretespace.ū) do u
+                    let discu = discretespace.discvars[u]
+                        solu = map(CartesianIndices(discu)) do I
+                            i = sym_to_index(discu[I], odesys)
+                            # Handle Observed
+                            if i !== nothing
+                                sol.u[i]
+                            else
+                                observed(sol, discu[I], :)
+                            end
+                        end
+                        out = zeros(T, length(sol.t), size(discu)...)
+                        for I in CartesianIndices(discu)
+                            out[:, I] .= solu[I]
+                        end
+                        u => out
+                    end
+                end)
+    # Build Interpolations
+    # TODO: Check if the grid is uniform and use the supported higher order interpolations
+    interp = Dict(map(keys(umap)) do k
```

Comment:
Why is this build here? Or, why not insert a generic function call here?

Reply:
@ChrisRackauckas Would a generic function call in SciMLBase call a function defined in MOL even if it does not use MOL, if the user was using MOL? Or do I misunderstand your point?

Where should this build go?